src/main/java/org/springframework/data/couchbase/core/convert/MappingCouchbaseConverter.java

```diff
@@ -501,33 +507,33 @@ public void doWithPersistentProperty(final CouchbasePersistentProperty prop) {
 				}
 
 				Object propertyObj = accessor.getProperty(prop, prop.getType());
-				if (null != propertyObj) {
-					if (prop.isAnnotationPresent(IdPrefix.class)) {
-						IdPrefix prefix = prop.findAnnotation(IdPrefix.class);
-						int order = prefix.order();
-						prefixes.put(order, convertToString(propertyObj));
-						return;
-					}
-
-					if (prop.isAnnotationPresent(IdSuffix.class)) {
-						IdSuffix suffix = prop.findAnnotation(IdSuffix.class);
-						int order = suffix.order();
-						suffixes.put(order, convertToString(propertyObj));
-						return;
-					}
-
-					if (prop.isAnnotationPresent(IdAttribute.class)) {
-						IdAttribute idAttribute = prop.findAnnotation(IdAttribute.class);
-						int order = idAttribute.order();
-						idAttributes.put(order, convertToString(propertyObj));
-					}
-
-					if (!conversions.isSimpleType(propertyObj.getClass())) {
-						writePropertyInternal(propertyObj, target, prop);
-					} else {
-						writeSimpleInternal(propertyObj, target, prop.getFieldName());
-					}
+
+				if (prop.isAnnotationPresent(IdPrefix.class)) {
```

Comment:
Please note: Annotation-based information should go inside of `PersistentProperty`. That will allow us in the future to react better to reflection-free metadata-caching.

Reply:
It's still inside doWithPersistentPropery().  I've just removed the if(propertyObj == null) { ... } guard around it, because it might need to be saved even if it is null.  That's what causes it to be out-dented.